redis/commands/graph/query_result.py

```diff
@@ -125,31 +134,32 @@ def parse_statistics(self, raw_statistics):
                 self.statistics[s] = v
 
     def parse_header(self, raw_result_set):
+        """
+        Parse the header of the result.
+        """
         # An array of column name/column type pairs.
         header = raw_result_set[0]
         return header
 
     def parse_records(self, raw_result_set):
-        records = []
+        """
+        Parses the result set and returns a list of records.
+        """
         result_set = raw_result_set[1]
-        for row in result_set:
-            record = []
-            for idx, cell in enumerate(row):
-                if self.header[idx][0] == ResultSetColumnTypes.COLUMN_SCALAR:  # noqa
-                    record.append(self.parse_scalar(cell))
-                elif self.header[idx][0] == ResultSetColumnTypes.COLUMN_NODE:  # noqa
-                    record.append(self.parse_node(cell))
-                elif (
-                    self.header[idx][0] == ResultSetColumnTypes.COLUMN_RELATION
-                ):  # noqa
-                    record.append(self.parse_edge(cell))
-                else:
-                    print("Unknown column type.\n")
-            records.append(record)
+        records = [
```

Comment:
IMHO this might be harder to read than just nested for loops. I don't think it's more efficient. WDYT? It's a style question mostly.

Reply:
Woah.. 50% faster to use comprehensions in my basic test case. Ignore me :D